rc/windowing/sway.kak

```diff
@@ -0,0 +1,66 @@
+provide-module sway %{
+
+# Ensure we're actually in Sway
+evaluate-commands %sh{
+    [ -z "${kak_opt_windowing_modules}" ] || 
+    [ -n "$SWAYSOCK" ] ||
+    echo 'fail SWAYSOCK is not set'
+}
+
+require-module 'wayland'
+
+define-command sway-focus-pid -hidden %{
+    evaluate-commands %sh{
+        # We need to walk our way up a tree of process IDs to find out which
+        # window a specific client is associated with. This can be further up
+        # than you might expect, as a window could be for a terminal emulator 
+        # that's running bash that's running kak.
+
+        focus_window_ancestor() {
+            # Get the container ID from the tree of windows
+            con_id=`swaymsg -t get_tree | jq ".. | (.nodes? // empty)[] | select(.pid == $1) | .id"`
```

Comment:
I think we can get rid of the jq dependency pretty easily with something like

```
if swaymsg -t get_tree | grep "\"pid\": $1"; then
    swaymsg [pid=$1] focus
else
   recursion to parent pid
fi
```

Reply:
Or even better:

```
pid=$1
while ! swaymsg [pid=$pid] focus; do
    pid=get parent pid, exit if invalid
done
```

Reply:
That's much better, thanks!